src/Control/Distributed/Process/Internal/Primitives.hs

```diff
@@ -982,7 +983,10 @@ expectTimeout n = receiveTimeout n [match return]
 spawnAsync :: NodeId -> Closure (Process ()) -> Process SpawnRef
 spawnAsync nid proc = do
   spawnRef <- getSpawnRef
-  sendCtrlMsg (Just nid) $ Spawn proc spawnRef
+  node     <- getSelfNode
+  if nid == node
+    then sendCtrlMsg Nothing $! Spawn (DS.force proc) spawnRef
```

Comment:
I propose we don't involve this change with broader concerns. This is a specific contribution which doesn't need further agreement on the future of `sendCtrlMsg`. The other alternative is to drop this PR entirely as the optimization is not essential for any purpose. It is a subproduct of fixing the d-p stack.

If we were to merge it:

> 1. should I move forcing value procedure to sendCtrlMsg itself (in Nothing branch)?

No. Keep it as it is and discuss further later.

> 2. should I use seq (encode x) x in order to force value in every place where we are sending (or should send but don't do it due to optimization)?

Do it only here. If there are other places where it is needed let us discuss it later.

> 3. should I introduce changes to any other use of sendCrtlMsg in this PR?

No, better to keep that for later.